src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "store" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class OpCacheAdapter implements AdapterInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private $file;
+    private $values;
+    private $createCacheItem;
+
+    /** @var AdapterInterface */
+    private $fallbackPool;
+
+    /**
+     * @return CacheItemPoolInterface
+     */
+    public static function createOnPhp7($file, CacheItemPoolInterface $fallbackPool)
+    {
+        // OPcache shared memory for arrays works only with PHP 7.0+
+        if (PHP_VERSION_ID >= 70000) {
+            if (!$fallbackPool instanceof AdapterInterface) {
+                $fallbackPool = new ProxyAdapter($fallbackPool);
+            }
+
+            return new static($file, $fallbackPool);
+        }
+
+        return $fallbackPool;
+    }
+
+    public function __construct($file, AdapterInterface $fallbackPool)
+    {
+        $this->file = $file;
+        $this->fallbackPool = $fallbackPool;
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+
+                return $item;
+            },
+            $this,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * Store a static array of cached values.
+     *
+     * @param array $values The cached values.
+     */
+    public function store(array $values)
+    {
+        if (file_exists($this->file)) {
+            if (!is_file($this->file)) {
+                throw new InvalidArgumentException(sprintf('Cache path exists and is not a file: %s', $this->file));
```

Comment:
Looking at this, `!is_file` doesn't mean `is_dir`, thus adding `is a directory` could be misleading (e.g. on Windows `filetype` can return `Unknown` for jonctions in some versions). Thus, I think we should keep the message asis...